src/adapters/cache.ts

```diff
@@ -1,27 +1,36 @@
-/* global Blob, DOMException, File, Response, caches, location */
-
-import { errors } from '../util.js'
+import { Adapter, FileSystemFileHandleAdapter, FileSystemFolderHandleAdapter, WriteChunk } from '../interfaces.js'
+import { errors, isChunkObject } from '../util.js'
 
 const { INVALID, GONE, MISMATCH, MOD_ERR, SYNTAX } = errors
 
 const DIR = { headers: { 'content-type': 'dir' } }
-const FILE = () => ({ headers: { 'content-type': 'file', 'last-modified': Date.now() } })
+const FILE = () => ({ headers: { 'content-type': 'file', 'last-modified': '' + Date.now() } })
+
+class Sink implements UnderlyingSink<WriteChunk> {
+  private _cache: Cache
+  private path: string
+  private size: number
+  private position: number
+  private file: File
 
-class Sink {
-  constructor (cache, path, file) {
+  constructor (cache: Cache, path: string, file: File) {
     this._cache = cache
     this.path = path
     this.size = file.size
     this.position = 0
     this.file = file
   }
 
-  write (chunk, c) {
-    if (typeof chunk === 'object') {
+  async write (chunk: WriteChunk) {
+    const [r] = await this._cache.keys(this.path)
+    if (!r) throw new DOMException(...GONE)
+
+    if (isChunkObject(chunk)) {
       if (chunk.type === 'write') {
-        if (Number.isInteger(chunk.position) && chunk.position >= 0) {
+        if (typeof chunk.position === 'number' && chunk.position >= 0) {
```

Comment:
Note to self: it's possible for position to be a string, web-idl stuff can make conversions based on some stuff like `symbol.toPrimitive` and other stuff

if we should do this kind of checking then chunk should be "normalized" before entering the Sink.